app/controllers/invitations_controller.rb

```diff
@@ -1,22 +1,19 @@
 class InvitationsController < Devise::InvitationsController
+  # PUT /resource/invitation
   def update
-    # Copied from https://github.com/scambra/devise_invitable/blob/master/app/controllers/devise/invitations_controller.rb
-    # And edited. This is a common devise pattern, similar to OmniauthCallbacksController.
     raw_invitation_token = update_resource_params[:invitation_token]
     self.resource = accept_resource
-    invitation_accepted = resource.errors.empty?
 
     yield resource if block_given?
 
     if invitation_accepted
-      resource.registered = true
-      resource.registered_at = Time.current
       if resource.class.allow_insecure_sign_in_after_accept
         flash_message = resource.active_for_authentication? ? :updated : :updated_not_active
         set_flash_message :notice, flash_message if is_flashing_format?
         resource.after_database_authentication
         sign_in(resource_name, resource)
-        redirect_to onboarding_path
+        resource.update!(registered_at: Time.current, registered: true)
+        respond_with resource, location: after_accept_path_for(resource)
```

Comment:
This mirrors the `devise_invitable` source code a bit more and works really well, IMO. 😄